src/memo/memo.mli

```diff
@@ -384,8 +396,11 @@ module Implicit_output : sig
   (** Register a new type of implicit output. *)
   val add : (module Implicit_output with type t = 'o) -> 'o t
 end
+with type 'a memo := 'a t
 
 module With_implicit_output : sig
+  type 'a memo
```

Comment:
You can now write `type 'a memo := 'a t` instead of an abstract type + substitution.

Reply:
Cool! Is this going to work in all OCaml versions we care about?

Reply:
Yes. We already use it in fiber btw

Reply:
I've pushed the change. Thanks!